CommonWallet/Classes/Modules/TransactionDetails/TransactionDetailsFactory.swift

```diff
@@ -0,0 +1,254 @@
+/**
+* Copyright Soramitsu Co., Ltd. All Rights Reserved.
+* SPDX-License-Identifier: GPL-3.0
+*/
+
+
+import Foundation
+import SoraFoundation
+
+public protocol WalletTransactionDetailsFactoryProtocol {
+    func createForm(from data: AssetTransactionData,
+                    type: WalletTransactionType,
+                    asset: WalletAsset) -> [WalletFormViewModelProtocol]
+}
+
+final class WalletTransactionDetailsFactory {
+    private let resolver: ResolverProtocol
+
+    init(resolver: ResolverProtocol) {
+        self.resolver = resolver
+    }
+
+    private func populateTransactionId(into viewModelList: inout [WalletFormViewModelProtocol],
```

Comment:
Looks like a builder, why inout here?

Reply:
The logic is quite specific to create separate builder. inout parameter allows changes in place instead of checking conditional add by checking return value.